packages/cdk/lib/construct/generic-agent-core.ts

```diff
@@ -79,26 +88,31 @@ export class GenericAgentCore extends Construct {
   }
 
   /**
-   * Deploy the generic AgentCore Runtime
+   * Deploy the generic AgentCore Runtime using L1 CDK Construct
    */
   private deployGenericRuntime(): { repository: Repository; imageUri: string } {
     const dockerImageAsset = this.createDockerImageAsset();
-    const { customResourceRole, agentCoreRuntimeRole } = this.createIamRoles();
-    const customResourceProvider =
-      this.createCustomResourceProvider(customResourceRole);
-
-    const customResource = this.createAgentCoreRuntime(
-      'GenericAgentCoreRuntime',
-      this.genericRuntimeConfig,
-      agentCoreRuntimeRole,
-      customResourceProvider,
-      dockerImageAsset.imageUri
-    );
+    const agentCoreRuntimeRole = this.createAgentCoreRuntimeRole();
```

Comment:
私の環境依存かもしれませんが、以下のエラーが発生していました。；

```bash
AgentCoreStack: creating CloudFormation changeset...
10:35:07 | CREATE_FAILED        | AWS::BedrockAgentCore::Runtime | GenericAgentCoreGe...eRuntimeL1B2F7F803
Resource handler returned message: "Invalid request provided: Access denied while validating ECR URI 'xxxx.dkr.ecr.us-east-1.amazonaws.com/c
dk-hnb659fds-container-assets-xxxxx-us-east-1:93a0bc6401c8848c3174de19d2010ba5897f4c44ebd6dd72ac3d3b9364165d52'. The execution role requires
permissions for ecr:GetAuthorizationToken, ecr:BatchGetImage, and ecr:GetDownloadUrlForLayer operations. (Service: BedrockAgentCoreControl, Status C
ode: 400, Request ID: b4d1d923-6c79-47b4-afd5-fe87c4a7fa98) (SDK Attempt Count: 1)" (RequestToken: edb004fc-9aa1-c040-b839-a4707f45513d, HandlerErro
rCode: InvalidRequest)
```

grantPull をつけて解消させました。vibe 解決させたのでもう少し良い改善方法があるかもしれませんが取り急ぎ共有です。

```suggestion
    const agentCoreRuntimeRole = this.createAgentCoreRuntimeRole();
    dockerImageAsset.repository.grantPull(agentCoreRuntimeRole);
```